### ISSUE TITLE
App Key Generation

### ISSUE THREAD

**Issue by digital801** [grokability/snipe-it#7881]
#### Expected Behavior (or desired behavior if a feature request)

run php artisan key:generate to generate app key in .env file.

-----

#### Actual Behavior

PHP Fatal error:  Uncaught Error: Class 'Illuminate\Foundation\Application' not found in /var/www/snipe-it/bootstrap/app.php:14
Stack trace:
#0 /var/www/snipe-it/artisan(18): require_once()
#1 {main}
  thrown in /var/www/snipe-it/bootstrap/app.php on line 14

-----

I have reinstalled composer, and also ran through and clear-compiled, etc. I read somewhere that composer should not be installed as root, admin, etc. I installed everything through the root account, is that the problem? If not installed from the root account, what account should it be installed from? Thanks

#### Please confirm you have done the following before posting your bug report:

- [ ] I have enabled debug mode 
- [X ] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Provide answers to these questions:

- Is this a fresh install or an upgrade? Fresh
- Version of Snipe-IT you're running: latest
- Version of PHP you're running: 7.3
- Version of MySQL/MariaDB you're running: latest
- What OS and web server you're running Snipe-IT on: Centos 7
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc): Git Clone
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error :NA
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error: NA
- If a stacktrace is provided in the error, include that too. NA
- Any errors that appear in your browser's error console. NA
- Confirm whether the error is reproducible on the demo: https://snipeitapp.com/demo. NA
- Include any additional information you can find in `storage/logs` and your webserver's logs. NA
- Include what you've done so far in the installation, and if you got any error messages along the way.NA
- Indicate whether or not you've manually edited any data directly in the database. NA

Please do not post an issue without answering the related questions above. If you have opened a different issue and already answered these questions, answer them again, once for every ticket. It will be next to impossible for us to help you.

https://snipe-it.readme.io/docs/getting-help

**Comment by snipe** [comment]
Did you run `composer install`?

**Comment by digital801** [comment]
Yes I did but I did the whole install, everything Apache, PHP, MariaDB from
the root user, is that the problem?

**Comment by snipe** [comment]
Did you *install composer*, or run `composer install`? Those are different things. (Never run composer as root btw)

**Comment by digital801** [comment]
When I run "composer install" I get the following:
[root@snipe snipe-it]# composer install
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for tecnickcom/tc-lib-barcode 1.15.16 -> satisfiable by tecnickcom/tc-lib-barcode[1.15.16].
    - tecnickcom/tc-lib-barcode 1.15.16 requires ext-bcmath * -> the requested PHP extension bcmath is missing from your system.

  To enable extensions, verify that they are enabled in your .ini files:
    - /etc/php.ini
    - /etc/php.d/20-bz2.ini
    - /etc/php.d/20-calendar.ini
    - /etc/php.d/20-ctype.ini
    - /etc/php.d/20-curl.ini
    - /etc/php.d/20-dom.ini
    - /etc/php.d/20-exif.ini
    - /etc/php.d/20-fileinfo.ini
    - /etc/php.d/20-ftp.ini
    - /etc/php.d/20-gd.ini
    - /etc/php.d/20-gettext.ini
    - /etc/php.d/20-iconv.ini
    - /etc/php.d/20-json.ini
    - /etc/php.d/20-ldap.ini
    - /etc/php.d/20-mbstring.ini
    - /etc/php.d/20-mysqlnd.ini
    - /etc/php.d/20-pdo.ini
    - /etc/php.d/20-phar.ini
    - /etc/php.d/20-simplexml.ini
    - /etc/php.d/20-sockets.ini
    - /etc/php.d/20-sqlite3.ini
    - /etc/php.d/20-tokenizer.ini
    - /etc/php.d/20-xml.ini
    - /etc/php.d/20-xmlwriter.ini
    - /etc/php.d/20-xsl.ini
    - /etc/php.d/30-mcrypt.ini
    - /etc/php.d/30-mysqli.ini
    - /etc/php.d/30-pdo_mysql.ini
    - /etc/php.d/30-pdo_sqlite.ini
    - /etc/php.d/30-wddx.ini
    - /etc/php.d/30-xmlreader.ini
    - /etc/php.d/40-zip.ini
  You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.
After that running that:
[root@snipe snipe-it]# php artisan key:generate
PHP Fatal error:  Uncaught Error: Class 'Illuminate\Foundation\Application' not found in /var/www/snipe-it/bootstrap/app.php:14
Stack trace:
#0 /var/www/snipe-it/artisan(18): require_once()
#1 {main}
  thrown in /var/www/snipe-it/bootstrap/app.php on line 14

**Comment by snipe** [comment]
> tecnickcom/tc-lib-barcode 1.15.16 requires ext-bcmath * -> the requested PHP extension bcmath is missing from your system.

That's your answer then. Install or enable the bcmath extension.

**Comment by digital801** [comment]
I apologize, I thought I had verified it was installed before I even posted that, but after checking loaded modules with "php -m", I saw it was not. After running:

yum update
yum install php-bcmath
composer install
composer update

Then verified it is loaded now with "php -m"  proceeded with php artisan key:generate and it generated successfully and is now installed in my env file. 

Thank you for helping and holding my hand a bit. Very much appreciated and I can now try to move on to the web install.
